lib/keyrod/cli.rb

```diff
@@ -100,5 +102,40 @@ def merge_config(options)
       ssl_params[:ca_path] = Keyrod::Settings[:'ca-dir'] if Keyrod::Settings[:'ca-dir']
       Keyrod::Settings[:ssl] = ssl_params
     end
+
+    def validate_config(options)
+      raise Keyrod::Errors::ParamsError, 'Refresh/access token required' unless options[:'access-token'] || options[:'refresh-token']
+      raise Keyrod::Errors::ParamsError, 'Use one of refresh/access token' if options[:'access-token'] && options[:'refresh-token']
+
+      return unless options[:'refresh-token']
+      validate_config_group options,
+                            ['oidc-site', 'client-id', 'client-secret'],
+                            '--oidc-site, --client-id, --client-secret are required with refresh token'
+    end
+
+    def validate_config_group(options, group, message)
+      return if group.all? { |option| options[option] }
+
+      raise Keyrod::Errors::ParamsError, message
+    end
+
+    def process_tokens
```

Comment:
Maybe try to refactor this method a bit. It took me a few minutes to understand the decision making process in there.